pkg/policymutation/policymutation.go

```diff
@@ -186,48 +184,62 @@ func generateRulePatches(policy kyverno.ClusterPolicy, controllers string, log l
 	for _, rule := range policy.Spec.Rules {
 		patchPostion := insertIdx
 
-		genRule = generateRuleForControllers(rule, controllers, log)
-		if reflect.DeepEqual(genRule, kyvernoRule{}) {
-			continue
-		}
+		convertToPatches := func(genRule kyvernoRule, patchPostion int) []byte {
+			operation := "add"
+			if existingAutoGenRule, alreadyExists := ruleMap[genRule.Name]; alreadyExists {
+				existingAutoGenRuleRaw, _ := json.Marshal(existingAutoGenRule)
+				genRuleRaw, _ := json.Marshal(genRule)
```

Comment:
Can we add error handling instead of skipping it?

Reply:
Yes we can, but [here](https://github.com/nirmata/kyverno/pull/1089/files#diff-0475b999b1b74960b8a0f4cf741f24a4R210) we are marshaling the entire Json Patches, if any error occurs, it should capture the error.